desktop/src/main.rs

```diff
@@ -111,20 +121,60 @@ fn main() {
 
     let opt = Opt::parse();
 
-    let ret = run_player(opt.input_path, opt.graphics, opt.power);
+    let ret = run_player(
+        opt.input_path,
+        opt.graphics,
+        opt.power,
+        opt.http_proxy,
+        opt.https_proxy,
+    );
 
     if let Err(e) = ret {
         eprintln!("Fatal error:\n{}", e);
         std::process::exit(-1);
     }
 }
 
+fn load_movie_from_path<P: AsRef<Path>>(
+    input_path: P,
+    http_proxy: Option<std::string::String>,
+    https_proxy: Option<std::string::String>,
+) -> Result<SwfMovie, Box<dyn std::error::Error>> {
+    let owned_path = input_path.as_ref();
+    let file_exists = owned_path.exists();
+    if !file_exists {
+        // Try parsing as URL instead
+        match Url::parse(owned_path.to_str().unwrap()) {
```

Comment:
Let's avoid the unwrap and panics here, this could be something like:
```
        let url = owned_path
            .to_str()
            .map(Url::parse)
            .ok_or("Input file does not exist and is not a network URL")?;
```

Reply:
I'll be honest, couldn't figure out how to get this to work.